### PR TITLE
Handle stop times enums & start ingesting stop times flex fields

### PR DESCRIPTION
**Pull request by lauriemerrell**
# Description
_Describe your changes and why you're making them. Please include the context, motivation, and relevant dependencies._

Result of investigating a question that @SorenSpicknall asked in Slack about why `trip_first_departure_time_sec` is sometimes null. I realized that we should probably be surfacing some of the `pickup` and `drop_off` enum values and flex-related fields which can help provide context for trips. 

So, this PR:

* Adds counts for each enum value for `stop_times` enums like `pickup/drop_off type` and `continuous pickup/drop_off` to `int_gtfs_schedule__stop_times_grouped`
* Ingests the new fields in `stop_times` that are part of [GTFS-Flex v2](https://github.com/MobilityData/gtfs-flex/blob/master/spec/reference.md)
* Adds summarized versions of that information (enums and flex) to `fct_daily_scheduled_trips` 
* Removes future service from most tables 
* Cleans up docs to reuse more between the stop times grouped and daily trips tables 

Progress towards #2574 but does not resolve it 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected) -- renames some columns, but [confirmed on Slack](https://cal-itp.slack.com/archives/C02H6JUSS9L/p1684183509863259) that that is ok
- [ ] Documentation

## How has this been tested?
_Include commands/logs/screenshots as relevant._

Confirmed that stop times grouped & fct daily trips and their respective entire lineages both run and pass all their tests. Did some spot checking that daily trips hasn't lost rows (except those being excluded for having no actual customer-facing service.)

## Post-merge follow-ups
_Document any actions that must be taken post-merge to deploy or otherwise implement the changes in this PR (for example, running a full refresh of some incremental model in dbt). If these actions will take more than a few hours after the merge or if they will be completed by someone other than the PR author, please create a dedicated follow-up issue and link it here to track resolution._

- [ ] No action required
- [x] Actions required (specified below)

* Full refresh stop times to get the new columns
* Update open data publishing data metadata -- separate ticket #2617

### CODE REVIEW

**Review by SorenSpicknall — APPROVED** [review]

File: warehouse/models/staging/gtfs/stg_gtfs_schedule__stop_times.sql [hunk 1]
```diff
@@ -23,7 +23,15 @@ stg_gtfs_schedule__stop_times AS (
...
-        SAFE_CAST({{ trim_make_empty_string_null('timepoint') }} AS INTEGER) AS timepoint
+        SAFE_CAST({{ trim_make_empty_string_null('timepoint') }} AS INTEGER) AS timepoint,
+        {{ trim_make_empty_string_null('start_pickup_drop_off_window') }} AS start_pickup_drop_off_window,
+        {{ trim_make_empty_string_null('end_pickup_drop_off_window') }} AS end_pickup_drop_off_window,
+        SAFE_CAST({{ trim_make_empty_string_null('mean_duration_factor') }} AS NUMERIC) AS mean_duration_factor,
+        SAFE_CAST({{ trim_make_empty_string_null('mean_duration_offset') }} AS NUMERIC) AS mean_duration_offset,
+        SAFE_CAST({{ trim_make_empty_string_null('safe_duration_factor') }} AS NUMERIC) AS safe_duration_factor,
+        SAFE_CAST({{ trim_make_empty_string_null('safe_duration_offset') }} AS NUMERIC) AS safe_duration_offset,
+        {{ trim_make_empty_string_null('pickup_booking_rule_id') }} AS pickup_booking_rule_id,
+        {{ trim_make_empty_string_null('drop_off_booking_rule_id') }} AS drop_off_booking_rule_id
```
Comment (atvaccaro) [inline warehouse/models/staging/gtfs/stg_gtfs_schedule__stop_times.sql:34]: integers? or are they already

File: warehouse/models/mart/gtfs_quality/fct_daily_reports_site_organization_scheduled_service_summary.sql [hunk 2]
```diff
@@ -32,7 +32,7 @@ fct_daily_reports_site_organization_scheduled_service_summary AS (
         SUM(n_trips) AS n_trips,
         MIN(first_departure_sec) AS first_departure_sec,
         MAX(last_arrival_sec) AS last_arrival_sec,
-        SUM(n_stop_times) AS n_stop_times,
+        SUM(num_stop_times) AS num_stop_times,
         SUM(n_routes) AS n_routes,
         COUNT(feed_key) AS feed_key_ct,
```
Comment (atvaccaro) [inline warehouse/models/mart/gtfs_quality/fct_daily_reports_site_organization_scheduled_service_summary.sql:37]: nit: thoughts on a follow-up to just change this everywhere?

File: warehouse/models/mart/gtfs/_mart_gtfs_fcts.yml [hunk 3]
```diff
@@ -452,17 +454,14 @@ models:
...
+      - name: num_distinct_stops_served
+        description: '{{ doc("column_num_distinct_stops_served") }}'
+      - name: num_stop_times
+        description: '{{ doc("column_num_stop_times") }}'
       - name: trip_first_departure_sec
         description: |
-          The number of seconds after midnight on `service_day` in `feed_timezone` at
-          which this trip's first stop departure occurred.
+          The number of seconds after 12 hours before noon (usually midnight) on `service_date`
+          in `feed_timezone` at which this trip's first stop departure occurred.
```
Comment (atvaccaro) [inline warehouse/models/mart/gtfs/_mart_gtfs_fcts.yml:464]: nit: this is hard to understand; is it `(noon - 12 hrs) + trip_first_departure_sec seconds`? Maybe you could include that (like the flex docs)

**Review by atvaccaro — APPROVED** [review]

File: warehouse/models/mart/gtfs_quality/fct_daily_reports_site_organization_scheduled_service_summary.sql (hunk 2 above)
Comment (lauriemerrell) [reply warehouse/models/mart/gtfs_quality/fct_daily_reports_site_organization_scheduled_service_summary.sql:37]: I'll revert this one and made separate follow up #2625

File: warehouse/models/mart/gtfs/_mart_gtfs_fcts.yml (hunk 3 above)
Comment (lauriemerrell) [reply warehouse/models/mart/gtfs/_mart_gtfs_fcts.yml:464]: Actually this is pretty standard verbiage in the `dims` yaml, would rather maintain consistency for now

**Pull request merged**